### ISSUE TITLE
panic on deletion

### ISSUE THREAD

**Issue by errordeveloper** [weaveworks/eksctl#756]
```
 [0] >> eksctl delete cluster scrumptious-rainbow-1556285024   
[ℹ]  using region us-west-2
[ℹ]  deleting EKS cluster "scrumptious-rainbow-1556285024"
[✔]  kubeconfig has been updated
[ℹ]  1 task: { delete cluster control plane "scrumptious-rainbow-1556285024" [async] }
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xe8 pc=0x1ffe2ca]

goroutine 30 [running]:
github.com/weaveworks/eksctl/pkg/cfn/manager.(*StackCollection).DeleteStackBySpec(0xc000b551a0, 0x0, 0x0, 0x45, 0xc0000fd710)
	/go/src/github.com/weaveworks/eksctl/pkg/cfn/manager/api.go:270 +0x3a
github.com/weaveworks/eksctl/pkg/cfn/manager.(*asyncTaskWithStackSpec).Do(0xc000950220, 0xc000098540, 0xc000098540, 0x1)
	/go/src/github.com/weaveworks/eksctl/pkg/cfn/manager/tasks.go:176 +0x33
github.com/weaveworks/eksctl/pkg/cfn/manager.doSingleTask(0xc0000984e0, 0x3b8a020, 0xc000950220)
	/go/src/github.com/weaveworks/eksctl/pkg/cfn/manager/tasks.go:185 +0xec
github.com/weaveworks/eksctl/pkg/cfn/manager.doSequentialTasks(0xc0000984e0, 0xc00029a0a0, 0x1, 0x1)
	/go/src/github.com/weaveworks/eksctl/pkg/cfn/manager/tasks.go:212 +0x50
created by github.com/weaveworks/eksctl/pkg/cfn/manager.(*TaskTree).DoAllSync
	/go/src/github.com/weaveworks/eksctl/pkg/cfn/manager/tasks.go:115 +0x246
 [2] >> eksctl version                                       
[ℹ]  version.Info{BuiltAt:"", GitCommit:"", GitTag:"0.1.29"}
 [0] >>
```
I think this could be to do with some of the nodegroups being an weird state, but nevertheless it needs looking into. Looks like it's fixed in master, but I cannot think of any recent change in this area.

**Comment by errordeveloper** [comment]
We do need to handle nil properly here, but also there is another issue.

As of #728, we broke backwards compatibility and there is no way for the user to know.

Namely: if a user creates a cluster using 0.1.30, and then tries to delete it using an older version (e.g. from a different machine), they will see this panic.

Here is why it occurs:
https://github.com/weaveworks/eksctl/blob/8c503bc3d3f8c50e84a25e3ca4d073d15845b6c4/pkg/cfn/manager/cluster.go#L48-L56

As I said, there are some improvements that can be made, but we cannot do so retrospectively... 

I think we do want to make sure to check if stack returned is not nil, that's for sure, but we also should figure out how to move forward with this.

One fix that I can think of would be to add the old tag back in, however the question is - how and when we maybe able to get rid of it safely?

**Comment by errordeveloper** [comment]
Closed via #758.
